### PR TITLE
feat(vue-demo-store): a11y improvements

### PR DESCRIPTION
**Pull request by mkucmus**
### Description

closes: #401 

solves all critical and serious issues. Some contrast issues cannot be tracked and found to fix (hidden DOM elements 🤔 probably). there are still some small issues in "alert" level mainly for redundant links for product (described in notes section). 

In order to fix many of contrast errors, I needed to tweak up the level of text colors, like for:
text-blue-500 -> text-blue-600, and so on. would be good to see if it goes aligned to our visual identity (if there is any).

### Notes
- dynamic content coming from API (CMS) cannot be fixed in the project. keep in mind that HTML from Shopping Experiences is not always valid in terms of accessibility rules.
- there is a problem with _redundant links_ ([see link purposes](https://webaim.org/standards/wcag/checklist#sc2.4.4) guideliness) with Product card -> one link for product's title, second for product image. we can rethink the structure of Product Card in order to place add to wishlist links in outer area to make possible wrapping in a link the image and title together.

### Used tools

- https://github.com/vue-a11y/vue-axe
- [Wave evaluation tool](https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh) (chrome extension)

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: templates/vue-demo-store/components/product/ProductCard.vue [hunk 1]
```diff
@@ -102,7 +102,6 @@ const srcPath = computed(() => {
       >
         <img
           ref="imageElement"
-          loading="lazy"
```
Comment (BrocksiNet) [inline templates/vue-demo-store/components/product/ProductCard.vue:105]: Lazy was not needed here?

File: templates/vue-demo-store/layouts/account.vue [hunk 2]
```diff
@@ -112,7 +112,7 @@ provide("swNavigation-footer-navigation", footerData);
                       @click="invokeLogout()"
                     >
                       <div i-carbon-logout text-xl inline-block />
-                      <span class="ml-3">Logout</span>
+                      <span class="ml-3 text-gray-700">Logout</span>
```
Comment (BrocksiNet) [inline templates/vue-demo-store/layouts/account.vue:115]: No translation needed here?

File: templates/vue-demo-store/i18n/de-DE/checkout.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
     "header": "Warenkorb",
     "empty": "Ihr Warenkorb ist leer",
     "close": "Panel schließen",
-    "logOut": "Sie können sich abmelden",
+    "logOut": "Sie können sich abmelden hier",
```
Comment (BrocksiNet) [inline templates/vue-demo-store/i18n/de-DE/checkout.json:11]: > Hier können Sie sich abmelden

Sounds better in german :)

File: templates/vue-demo-store/nuxt.config.ts [hunk 4]
```diff
@@ -10,6 +10,7 @@ export default defineNuxtConfig({
       shopwareEndpoint: "",
     },
     public: {
+      env: process.env.NODE_ENV || "development",
```
Comment (BrocksiNet) [inline templates/vue-demo-store/nuxt.config.ts:13]: Needed for?

**Review by BrocksiNet — APPROVED** [review]

File: templates/vue-demo-store/components/product/ProductCard.vue (hunk 1 above)
Comment (mkucmus) [reply templates/vue-demo-store/components/product/ProductCard.vue:105]: as it's used in product listing, mostly, there is no need to have it loaded in a lazy way, due to possibly low CLS score.

File: templates/vue-demo-store/i18n/de-DE/checkout.json (hunk 3 above)
Comment (mkucmus) [reply templates/vue-demo-store/i18n/de-DE/checkout.json]: 🇩🇪 💯

File: templates/vue-demo-store/nuxt.config.ts (hunk 4 above)
Comment (mkucmus) [reply templates/vue-demo-store/nuxt.config.ts]: removed 👍🏼

File: templates/vue-demo-store/layouts/account.vue (hunk 2 above)
Comment (mkucmus) [reply templates/vue-demo-store/layouts/account.vue]: you are totally right, I added translations to the langs' snippets 👍🏼

**Review by patzick — APPROVED** [review]

**Pull request merged**